packages/web-components/src/component-mixins/media-query/media-query.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { Constructor } from '../../globals/defs';
+
+/**
+ * @param Base The base class.
+ * @returns A mix-in implementing the logic for performing an action when the
+ * viewport width crosses over a configurable breakpoint.
+ */
+const MediaQueryMixin = <T extends Constructor<HTMLElement>>(Base: T) => {
+  abstract class MediaQueryMixinImpl extends Base {
+    /**
+     * A MediaQueryList object that indicates when we've crossed over the
+     * breakpoint threshold.
+     *
+     * Magic Number: The default, 1056px, matches 'carbon--breakpoint(lg)' mixin.
+     *
+     * @see https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList
+     */
+    _mediaQuery: MediaQueryList = window.matchMedia(`(max-width: 1056px)`);
```

Comment:
We should get this breakpoint from the @carbon/layout package

Reply:
See [the table-of-contents component](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/3c07ce0ce07063bc46b07bae36d287002713a93c/packages/web-components/src/components/table-of-contents/table-of-contents.ts#L42) for an example of this